app/src/main/java/com/example/myapplication/utils/NewsAdapter.kt

```diff
@@ -0,0 +1,48 @@
+package com.example.myapplication.utils
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.example.myapplication.R
+
+class NewsAdapter(private val dataSet: List<String>) :
```

Comment:
For now its okay to have a list of strings in your adapter, but you may use a better object to use the adapter. Since its a new adapter, maybe receiving a list of news would be more consistent :)